analyzers/src/SonarAnalyzer.CFG/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs

```diff
@@ -528,6 +534,12 @@ private Block BuildPatternExpression(PatternSyntaxWrapper patternSyntaxWrapper,
             throw new NotSupportedException($"{patternSyntaxWrapper.SyntaxNode.Kind()}");
         }
 
+        private static bool IsTooComplex(SyntaxNode node)
```

Comment:
given you only use it in `BuildExpression`, can't this be a static local function?

(this class already has lots of methods - and you group this private static method inside the region `Top level Build` - you may as well keep it as a local function)

Reply:
Method `BuildExpression` itself has almost 300 LOC. I don't want to add local function there.